### PR TITLE
[doc] Do some polishing in IDEs section

### PR DESCRIPTION
**Pull request by andresdelfino**
* Update link to pywin32
* Update link to ActivePython
* Update link to Komodo IDE
* Change reference from pydb (Python 2) to trepan3k (successor for Python 3, no longer compatible with DDD)
* Add Visual Studio Code (a popular choice)

### CODE REVIEW

File: Doc/faq/programming.rst [hunk 1]
```diff
@@ -28,27 +28,27 @@ The IDLE interactive development environment, which is part of the standard
 Python distribution (normally available as Tools/scripts/idle), includes a
 graphical debugger.
 
-PythonWin is a Python IDE that includes a GUI debugger based on pdb.  The
+Pythonwin is a Python IDE that includes a GUI debugger based on pdb.  The
```
Comment (orsenthil) [inline Doc/faq/programming.rst]: The correct term is `PythonWin`

File: Doc/faq/programming.rst [hunk 2]
```diff
@@ -28,27 +28,27 @@ The IDLE interactive development environment, which is part of the standard
...
 Pythonwin debugger colors breakpoints and has quite a few cool features such as
-debugging non-Pythonwin programs.  Pythonwin is available as part of the `Python
-for Windows Extensions <https://sourceforge.net/projects/pywin32/>`__ project and
-as a part of the ActivePython distribution (see
-https://www.activestate.com/activepython\ ).
+debugging non-Pythonwin programs.  Pythonwin is available as part of
```
Comment (orsenthil) [inline Doc/faq/programming.rst]: `PythonWin`

File: Doc/faq/programming.rst (hunk 1 above)
Comment (andresdelfino) [reply Doc/faq/programming.rst]: Done!

File: Doc/faq/programming.rst (hunk 2 above)
Comment (andresdelfino) [reply Doc/faq/programming.rst]: Done!

**Review by orsenthil — APPROVED** [review]

**Pull request merged**